### ISSUE TITLE
support for project references?

### ISSUE THREAD

**Issue by JasonKleban** [whitecolor/ts-node-dev#239]
## Issue description

On the latest ts-node-dev, it seems to behave akin to `tsc --watch -p .` because I get errors about project reference imported modules not being found.  Is there a way to have it operate akin to `tsc --build --watch` instead so that project references are supported?  I don't see any mention of it in the docs or issues.  Thanks!

## Context

*OS version (is it docker or host?), ts-node-dev version*

Windows 10, tsnd 1.1.1

**Comment by whitecolor** [comment]
Did you try to run with `--files` flag?

**Comment by JasonKleban** [comment]
https://github.com/TypeStrong/ts-node#help-my-types-are-missing

  > TypeScript Node does not use files, include or exclude, by default. ... If you must use files, include, or exclude, enable `--files` flags

So I tried `tsnd --files -- ./src` but it doesn't seem to make any difference.  My .tsconfigs don't use `files: []` at all anyway.

`ts-node`'s support for `--build` is unclear to me.  I found these old issues, but it feels like they might just be stale at this point:

 * https://github.com/TypeStrong/ts-node/issues/817 open issue about `--incremental` support
 * https://github.com/TypeStrong/ts-node/issues/817#issuecomment-546558202 2019 call for PRs, but unfortunately lacking any guidance or outline of a recommended direction.
 * https://github.com/TypeStrong/ts-node/issues/817#issuecomment-578346213 confusing 2019 comment about there being support for `--incremental` though this still isn't documented
 * https://github.com/TypeStrong/ts-node/pull/895 ts-node supposed support for --incremental (??), though this doesn't appear to be documented and I haven't gotten it to work
 * https://github.com/microsoft/TypeScript/issues/31849 about documenting the tsc api and some confusion about --incremental vs --build
 * https://github.com/microsoft/TypeScript-wiki/pull/225 sleeper PR of documentation, with possible use of unstable/abandoned tsc API

I'm gathering that there's an important difference - incremental api even applicable within a single project, whereas `--build` adds to that the following of project references.

**Comment by whitecolor** [comment]
Does running with `ts-node` work?

**Comment by JasonKleban** [comment]
I don't think so.  I guess I'm just looking for clarification on what can be expected to work these days.  Perhaps there's a flag that you or others are yourselves using successfully that I'm missing.  Or you perhaps have a nuanced understanding of why this can't work, or an outline of particularly what change in ts-node or tsc's api would have to happen before this usecase can be supported.

**Comment by wclr** [comment]
> because I get errors about project reference

I even didn't see your errors.

**Comment by JasonKleban** [comment]
The errors are the same as when I (incorrectly) try `tsc --project .` instead of `tsc --build`.  It complains of not being able to find modules that are project references.

```
error TS2307: Cannot find module 'host-common' or its corresponding type declarations.

1 import { foo as fooCommon } from "host-common";
```

**Comment by wclr** [comment]
I'm not sure why tsc can not see this package types

**Comment by JasonKleban** [comment]
Do you mean that you expect this to work, and that it might just be something configured incorrectly with my project?  Or are you saying that you don't know why `tsc` requires `--build` mode to understand project references?   (I don't know that either, but I've seen hints in documentation that it is for backward compatibility with older typescript projects)

**Comment by wclr** [comment]
There is something with your project configuration I believe. You may try to make a simple repro example.

**Comment by Aveyder** [comment]
For those facing the same problem, this is primarily related to `ts-node` see https://github.com/TypeStrong/ts-node/issues/897 (looks like it doesn't support `tsc -b` flag at the moment). In the mentioned issue some workarounds can be found:

1. Using `concurrently` with `tsc -b -w` & `nodemon`: 
https://github.com/TypeStrong/ts-node/issues/897#issuecomment-584084863
2. Using `tsc-watch` & `node`
https://github.com/TypeStrong/ts-node/issues/897#issuecomment-647035509
